src/esp/physics/PhysicsManager.cpp

```diff
@@ -88,21 +91,27 @@ int PhysicsManager::addObject(const int objectLibIndex,
 
 int PhysicsManager::addObject(const std::string& configFile,
                               DrawableGroup* drawables,
+                              scene::SceneNode* attachmentNode,
```

Comment:
May I understand why this argument is a must have?
Can we just create a scene object within addObject(...), and thus hide this detail to the user?
I did not see any usage case in this PR that leverage this argument. The only caller (in Simulator.cpp) just passed a nullptr.

Did I miss the big picture here?

Reply:
I pushed this into python and wrote a test case in test_physics.py to demonstrate the usage. This PR is really to address the visual/simulated agent problems. An Agent `SceneNode` can be passed to `addObject` to use that node instead of generating a new one.